sormas-ui/src/main/java/de/symeda/sormas/ui/samples/SampleController.java

```diff
@@ -205,19 +207,23 @@ public PathogenTestForm addPathogenTestComponent(
 		};
 		sampleComponent.addCommitListener(savePathogenTest);
 		// Discard button configuration
-		Button discardButton = ButtonHelper.createButton(I18nProperties.getCaption(Captions.pathogenTestRemove));
-		VerticalLayout buttonLayout = new VerticalLayout(discardButton);
-		buttonLayout.setComponentAlignment(discardButton, Alignment.TOP_LEFT);
-		// add the discard button above the overall discard and commit buttons
-		sampleComponent.addComponent(buttonLayout, sampleComponent.getComponentCount() - 1);
-		discardButton.addClickListener(o -> {
-			sampleComponent.removeComponent(horizontalRule);
-			sampleComponent.removeComponent(buttonLayout);
-			sampleComponent.removeComponent(pathogenTestForm);
-			sampleComponent.removeFieldGroups(pathogenTestForm.getFieldGroup());
-			sampleComponent.removeCommitListener(savePathogenTest);
-			pathogenTestForm.discard();
-		});
+		if ((UserProvider.getCurrent().hasUserRight(UserRight.PATHOGEN_TEST_DELETE) && (nonNull(pathogenTest)))
```

Comment:
This does not reliably hide the button for users who may not delete existing pathogen tests. In the [LabMessageProcessingUIHelper.getSampleEditComponent(...)](https://github.com/hzi-braunschweig/SORMAS-Project/blob/c045b5e6009cf28430169be2ce25de7daa054b79/sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/processing/LabMessageProcessingUIHelper.java#L104), existing and new pathogen tests are added as non-null parameters ([here](https://github.com/hzi-braunschweig/SORMAS-Project/blob/c045b5e6009cf28430169be2ce25de7daa054b79/sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/processing/LabMessageProcessingUIHelper.java#L120) and [here ](https://github.com/hzi-braunschweig/SORMAS-Project/blob/c045b5e6009cf28430169be2ce25de7daa054b79/sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/processing/LabMessageProcessingUIHelper.java#L151)).